packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1764,6 +1765,12 @@ function resetChildLanes(completedWork: Fiber) {
   }
 
   let newChildLanes = NoLanes;
+  let subtreeTag = NoEffect;
+  let childrenDidNotComplete = false;
+
+  const wasFiberCloned =
+    completedWork.alternate === null ||
+    completedWork.child !== completedWork.alternate.child;
```

Comment:
I believe I see one potential over-traversal problem. Although I'm not sure if it affects this branch, it does affect the follow up branch (passive effects): If a component gets a state update, but its siblings don't, we still end up traversing the sibling path when running passive effects which leads us to over-count commit times. Fortunately the Profiler tests are pretty good and caught this.

Maybe using an explicit flag to mark bailed-out-fibers would (like we talked about on Tuesday) could help with that too.

Reply:
It should be fine that the siblings are traversed in that case as long as they are also cloned, because if they are cloned then we should be able to clear the effect tags in `createWorkInProgressFiber`.

I do think the explicit flag approach is worth approaching, though.

Reply:
Gotcha. I'm seeing a few failures that are being caused by over-traversal during passive effects phase, so something doesn't seem to be getting reset correctly. Will keep digging in.

Tried the explicit flag approach initially and it failed more things but I'm probably missing some of the bailout spots.

Reply:
Explicit bailout flag approach has been implemented in 2nd part PR, #19322